python/pyspark/sql/tests.py

```diff
@@ -1246,6 +1246,31 @@ def test_struct_type(self):
         with self.assertRaises(TypeError):
             not_a_field = struct1[9.9]
 
+    def test_parse_datatype_string(self):
+        from pyspark.sql.types import _all_atomic_types, _parse_datatype_string
+        for k, t in _all_atomic_types.items():
+            if t != NullType:
```

Comment:
So, if I haven't missed anything, this PR drops the support the type parsing `null`. I guess it is almost seldom that we explicitly set the type with `null`. Also, IIRC, we will support `NullType` via `void` (SPARK-20680) soon as a workaround.